hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -793,8 +793,15 @@ public HRegion(final HRegionFileSystem fs, final WAL wal, final Configuration co
       throw new IllegalArgumentException(MEMSTORE_FLUSH_PER_CHANGES + " can not exceed "
           + MAX_FLUSH_PER_CHANGES);
     }
-    this.rowLockWaitDuration = conf.getInt("hbase.rowlock.wait.duration",
-                    DEFAULT_ROWLOCK_WAIT_DURATION);
+    int tmpRowLockDuration = conf.getInt("hbase.rowlock.wait.duration",
+        DEFAULT_ROWLOCK_WAIT_DURATION);
+    if (tmpRowLockDuration <= 0) {
```

Comment:
nit: We are not expecting this condition for upgrade to 3.0.0 right? We can either roll this out to branch-2.2+ except master or we can remove `behavior of hbase versions older than 1.4` from log msg at least in master branch to treat this as generic condition regardless of how HBase version<1.4 treated negative rowLockDuration. For branch-2.2+, log msg as is should be good.
Thought?

Reply:
I dunno. If the reason we're treating it as 1ms, rather than e.g. refusing to open the region, is because in old hbase versions you could rely on this behavior then that's no less true in hbase 4 then in hbase 1.7.0.

Reply:
We could remove mention of the reasoning about old versions of hbase all the time though and just say we are doing it since not doing so will mean the region won't allow row locks?

Reply:
Sure, removing the mention of old HBase version behaviour sounds good 👍